src/Relude/Extra/Foldable.hs

```diff
@@ -38,3 +40,15 @@ foldlSC f = flip $ foldr go id
         Left l  -> l
         Right r -> k r
 {-# INLINE foldlSC #-}
+
+{- | Compute average of 'Foldable'
+-}
+average :: (Foldable f, Fractional a) => f a -> Maybe a
```

Comment:
Let's specify explicit order of arguments here. I imagine, people will specify values type more often than the container type with TypeApplications.

```suggestion
average :: forall a f . (Foldable f, Fractional a) => f a -> Maybe a
```